src/transformers/models/moonshine/modular_moonshine.py

```diff
@@ -1109,22 +1098,106 @@ def forward(
             encoder_attentions=outputs.encoder_attentions,
         )
 
-    def generate(self, *args, **kwargs):
-        # TODO: @eustlb do it rather with a custom logits processor
-        token_limit_factor = 6.5 / 16000.0  # Maximum of 6.5 tokens per second
-        if kwargs.get("max_new_tokens") is None and kwargs.get("max_length") is None:
-            if kwargs.get("attention_mask") is not None:
-                seq_lens = kwargs["attention_mask"].sum(dim=-1)
+    def generate(
+        self,
+        input_values: Optional[torch.FloatTensor] = None,
+        infer_max_length_from_input: bool = True,
+        tokens_per_second: float = None,
+        **generate_kwargs,
+    ):
+        """
+        Overrides GenerationMixin's [`~generation.GenerationMixin.generate`] method to add a `PerBatchIndexMaxLengthLogitsProcessor` by default.
+        Indeed Moonshine model uses per-batch-index max length that are inferred from the input audio length.
```

Comment:
```suggestion
        Overrides GenerationMixin's [`~generation.GenerationMixin.generate`] method to add a `PerBatchIndexMaxLengthLogitsProcessor` as default because Moonshine uses per-batch-index max length that is inferred from the input audio length.
```
```